app/controllers/shared/api.php

```diff
@@ -59,13 +59,16 @@
         }
 
         $abuse = new Abuse($timeLimit);
+        $remaining = $timeLimit->remaining();
+        $limit = $timeLimit->limit();
+        $time = (new DateTime($timeLimit->time()))->getTimestamp() + $route->getLabel('abuse-time', 3600);
 
-        if ($timeLimit->limit() && ($timeLimit->remaining() < $closestLimit || is_null($closestLimit))) {
-            $closestLimit = $timeLimit->remaining();
+        if ($limit && ($remaining < $closestLimit || is_null($closestLimit))) {
+            $closestLimit = $remaining;
             $response
-                ->addHeader('X-RateLimit-Limit', $timeLimit->limit())
-                ->addHeader('X-RateLimit-Remaining', $timeLimit->remaining())
-                ->addHeader('X-RateLimit-Reset', $timeLimit->time() + $route->getLabel('abuse-time', 3600))
+                ->addHeader('X-RateLimit-Limit', $limit)
+                ->addHeader('X-RateLimit-Remaining', $remaining)
+                ->addHeader('X-RateLimit-Reset', $time)
```

Comment:
I think this is going to cause an exception when it tries to convert the `DateTime` to string.